### PR TITLE
Bob 80  code aufraeumen

### PR DESCRIPTION
**Pull request by DonTars**

### CODE REVIEW

File: Assets/Scripts/Effects/SpawnParticleEffect.cs [hunk 1]
```diff
@@ -1,50 +1,44 @@
...
             }
         }
 
         private void OnTriggerEnter2D(Collider2D col)
         {
             if (!onDestroy && onCollision && !useTag)
-            { 
-                SpawnEffect();  
+            {
+                SpawnEffect();
```
Comment (SomethingWittyWithATechAspect) [inline Assets/Scripts/Effects/SpawnParticleEffect.cs:26]: return;

File: Assets/Scripts/Effects/SpawnParticleEffect.cs [hunk 2]
```diff
@@ -1,50 +1,44 @@
...
             }
-            if (!onDestroy && onCollision && useTag)
+
+            if (onDestroy || !onCollision || !useTag) return;
```
Comment (SomethingWittyWithATechAspect) [inline Assets/Scripts/Effects/SpawnParticleEffect.cs:29]: should be first line - it easier to read if  "if return" are at the beginning

File: Assets/Scripts/Effects/SpawnParticleEffect.cs [hunk 3]
```diff
@@ -1,50 +1,44 @@
...
+            if (col.CompareTag(tag))
```
Comment (SomethingWittyWithATechAspect) [inline Assets/Scripts/Effects/SpawnParticleEffect.cs:30]: useTag && col.CompareTag or comment that is is implied that useTag is true

File: Assets/Scripts/MapTools/Helper/SmallFloorConnector.cs [hunk 4]
```diff
@@ -1,14 +1,15 @@
...
-namespace MapTools
+namespace MapTools.Helper
 {
     /// <summary>
     /// Removes all small floors on the given map
     /// </summary>
     public class SmallFloorConnector
     {
         private (bool wall, int floorID)[,] _floorMatrix;
+        // ReSharper disable once HeapView.ObjectAllocation.Evident
```
Comment (SomethingWittyWithATechAspect) [inline Assets/Scripts/MapTools/Helper/SmallFloorConnector.cs:12]: remove or disable in global conf

File: Assets/Scripts/MapTools/Door.cs [hunk 5]
```diff
@@ -1,6 +1,7 @@
 ﻿using UnityEditor.PackageManager;
 using UnityEngine;
 using UnityEngine.SceneManagement;
+using Logger = BoBLogger.Logger;
```
Comment (SomethingWittyWithATechAspect) [inline Assets/Scripts/MapTools/Door.cs:4]: at other places using static BobLogger.logger is used 
pls only use one variant

Comment (SomethingWittyWithATechAspect) [reply Assets/Scripts/MapTools/Door.cs:4]: see spawnobject as example

File: Assets/Scripts/ShooterController/AutoShooter.cs [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+
+        private static GameObject FindClosest(GameObject origin, string find)
+        {
+            GameObject closest = null;
+            var distanceToClosest = Mathf.Infinity;
+            var allEnemy = GameObject.FindGameObjectsWithTag(find);
+            foreach (var currentEnemy in allEnemy)
+            {
+                var currentDistance = (currentEnemy.transform.position - origin.transform.position).sqrMagnitude;
+                if (!(currentDistance < distanceToClosest)) continue;
```
Comment (SomethingWittyWithATechAspect) [inline Assets/Scripts/ShooterController/AutoShooter.cs:42]: still no fan of continue

File: Assets/Scripts/UI/Hud/ResourceListUIScript.cs [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+                var resUi = this.transform.Find(vResource.Key)?.gameObject;
+                if (resUi == null)
+                {
+                    resUi = Instantiate(resourceUi, this.transform);
+                    resUi.name = vResource.Key;
+                }
+
+                foreach (var child in resUi.GetComponentsInChildren<Text>())
+                {
+                    child.text = child.name switch
```
Comment (SomethingWittyWithATechAspect) [inline Assets/Scripts/UI/Hud/ResourceListUIScript.cs:25]: for me this is the ugliest way to write a switch

File: Assets/Scripts/Effects/SpawnParticleEffect.cs [hunk 8]
```diff
@@ -1,50 +1,44 @@
...
         {
             if (!spawnParticleEffect.CompareTag("Circle"))
-            { 
-                SpawnObject.Spawn(spawnParticleEffect, transform.localPosition, transform.rotation);
+            {
+                var tempTransform = transform;
+                SpawnObject.Spawn(spawnParticleEffect, tempTransform.localPosition, tempTransform.rotation);
             }
             else
             {
```
Comment (SomethingWittyWithATechAspect) [inline Assets/Scripts/Effects/SpawnParticleEffect.cs:44]: creates null pointer 
NullReferenceException: Object reference not set to an instance of an object
MapTools.SpawnObject.GetNewObject (UnityEngine.GameObject obj, UnityEngine.Vector3 location) (at Assets/Scripts/MapTools/SpawnObject.cs:37)
MapTools.SpawnObject.Spawn (UnityEngine.GameObject obj, UnityEngine.Vector3 location) (at Assets/Scripts/MapTools/SpawnObject.cs:26)
Effects.SpawnParticleEffect.SpawnEffect () (at Assets/Scripts/Effects/SpawnParticleEffect.cs:45)
Effects.SpawnParticleEffect.OnDestroy () (at Assets/Scripts/Effects/SpawnParticleEffect.cs:18)

**Review by SomethingWittyWithATechAspect — CHANGES_REQUESTED** [review]
player Shooting speed is still to hihg
Tower still cost 1 resp. 2 wood

All in all great job

File: Assets/Scripts/Effects/SpawnParticleEffect.cs (hunk 8 above)
Comment (spirit126) [reply Assets/Scripts/Effects/SpawnParticleEffect.cs:44]: fixed

File: Assets/Scripts/UI/Hud/ResourceListUIScript.cs (hunk 7 above)
Comment (spirit126) [reply Assets/Scripts/UI/Hud/ResourceListUIScript.cs:25]: a switch expression is a short form of switches, and the way to go

File: Assets/Scripts/ShooterController/AutoShooter.cs (hunk 6 above)
Comment (spirit126) [reply Assets/Scripts/ShooterController/AutoShooter.cs:41]: unity coding guidelines ...

File: Assets/Scripts/MapTools/Door.cs (hunk 5 above)
Comment (spirit126) [reply Assets/Scripts/MapTools/Door.cs:3]: fixt

File: Assets/Scripts/Effects/SpawnParticleEffect.cs (hunk 3 above)
Comment (spirit126) [reply Assets/Scripts/Effects/SpawnParticleEffect.cs:30]: reworked

File: Assets/Scripts/Effects/SpawnParticleEffect.cs (hunk 2 above)
Comment (spirit126) [reply Assets/Scripts/Effects/SpawnParticleEffect.cs:29]: reworked

File: Assets/Scripts/Effects/SpawnParticleEffect.cs (hunk 1 above)
Comment (spirit126) [reply Assets/Scripts/Effects/SpawnParticleEffect.cs:26]: reworked

File: Assets/Scripts/Attributes/EffectList.cs [hunk 9]
```diff
@@ -1,51 +1,54 @@
...
+            var toRemove = new List<T>();
+            foreach (var eff in _effects)
             {
-                if (currentHighestValue < eff.getEffect())
+                eff.DecreaseEffectTimeBy(time);
+                if (eff.GetEffectTime() <= 0)
+                {
+                    toRemove.Add(eff);
+                }
+                else
```
[the remark this answers is not in the document]
Comment (spirit126) [reply Assets/Scripts/Attributes/EffectList.cs:41]: reworked

File: Assets/Scripts/Effects/SpawnParticleEffect.cs (hunk 3 above)
Comment (SomethingWittyWithATechAspect) [reply Assets/Scripts/Effects/SpawnParticleEffect.cs]: looks nice

**Review by SomethingWittyWithATechAspect — APPROVED** [review]

**Pull request merged**